lib_nbgl/include/nbgl_use_case_v2.h

```diff
@@ -0,0 +1,235 @@
+ * @param content content to fill
+ */
+typedef void (*nbgl_contentCallback_t)(uint8_t contentIndex, nbgl_content_t *content);
+
+typedef struct {
+    bool callback_call_needed;  ///< indicates whether contents should be retrieved using
+                                ///< contentsList or contentGetterCallback
+    union {
+        const nbgl_content_t *contentsList;  ///< array of nbgl_content_t (nbContents items).
+        nbgl_contentCallback_t
+            contentGetterCallback;  ///< function to call to retrieve a given content
+    };
+    uint8_t nbContents;  ///< number of contents
+} nbgl_genericContents_t;
+
+/**
+ * @brief prototype of content navigation callback function
+ * @param contentsIndex contents index (0->(nbContents-1)) that is needed by the lib
+ * @param contents content to fill
+ * @return true if the page content is valid, false if no more page
+ */
+typedef bool (*nbgl_genericContentsCallback_t)(uint8_t contentsIndex, nbgl_genericContents_t *contents);
+
+
+/*
+ * Generic Nav, should be used only in non-nominal cases (for example sub-settings levels)
+ */
+
+// TODO: API to polish, but main idea is to specify an array of contents and a navigation configuration
+// like touchable, progressIndicator and all...
+// this should be used in apps:
+// - that were previously using nbgl_useCaseSettings for that
+// - that have custom content to display that doesn't fit classical cases?
+void nbgl_useCaseGenericNav(const char                   *title,
+                            const nbgl_genericContents_t *contents,
+                            uint8_t                       initPage,
+                            bool                          touchable,
+                            bool                          progressIndicator,
+                            nbgl_callback_t               quitCallback);
+
+
+/*
+ * Home and settings
+ */
+
+void nbgl_useCaseGenericSettings(const char                   *appName,
+                                 uint8_t                       initPage,
+                                 const nbgl_genericContents_t *settingContents,
+                                 const nbgl_contentInfoList_t *infosList,
+                                 nbgl_callback_t               quitCallback);
+
+void nbgl_useCaseHomeAndSettings(const char                   *appName,
+                                 const nbgl_icon_details_t    *appIcon,
+                                 const char                   *tagline,
+                                 const uint8_t                *initSettingPage, // Set to NULL if settings shouldn't be displayed right away
+                                 const nbgl_contentInfoList_t *infosList,
+                                 const nbgl_genericContents_t *settingContents,
+                                 nbgl_callback_t               quitCallback);
+
+void nbgl_useCaseHomeExtAndSettings(const char                   *appName,
```

Comment:
Could we avoid this dedicated API, considering the parameters actionText and actionCallback could simply be `NULL`?